### PR TITLE
fix(web.domain.webhosting): display error message with details

### PR DESCRIPTION
**Pull request by marie-j**
Issue brought with DTRSD-6830

### CODE REVIEW

File: packages/manager/apps/web/client/app/domain/webhosting/order/steps/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -105,7 +105,7 @@
   "domain_webhosting_order_payment_duration": "Durée :",
   "domain_webhosting_order_duration_P1Y": "12 mois",
   "domain_webhosting_order_payment_choice": "Afin de confirmer votre demande, veuillez choisir votre moyen de paiement, et accepter les conditions d'utilisation.",
-  "domain_webhosting_order_payment_get_checkout_error": "Une erreur est survenue lors de la récupération des informations de votre panier.",
+  "domain_webhosting_order_payment_get_checkout_error": "Une erreur est survenue lors de la récupération des informations de votre panier: {{ message }}",
```
Comment (antleblanc) [inline packages/manager/apps/web/client/app/domain/webhosting/order/steps/translations/Messages_fr_FR.json]:
```suggestion
  "domain_webhosting_order_payment_get_checkout_error": "Une erreur est survenue lors de la récupération des informations de votre panier : {{ message }}",
```

File: packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js [hunk 2]
```diff
@@ -78,7 +77,9 @@ const resolve = {
         }),
       )
       .catch(() => {
-        displayErrorMessage('domain_webhosting_order_payment_checkout_error');
+        displayErrorMessage(
+          $translate.instant('domain_webhosting_order_payment_checkout_error'),
```
Comment (frenautvh) [inline packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js]: If you are not providing an error message then the sentence will end with "panier :" right?
Maybe you should not modify the translation and simply add a new paragraph below with the error message

Comment (cbourgois) [reply packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js]: Or maybe we should have 2 translations : first with error message (`domain_webhosting_order_payment_get_checkout_error`) and another one without (`domain_webhosting_order_payment_get_checkout_error_unknown`)

Comment (marie-j) [reply packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js]: @frenautvh not for that translation but for the other one. I'll update

File: packages/manager/apps/web/client/app/domain/webhosting/order/steps/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -105,9 +105,11 @@
   "domain_webhosting_order_payment_duration": "Durée :",
   "domain_webhosting_order_duration_P1Y": "12 mois",
   "domain_webhosting_order_payment_choice": "Afin de confirmer votre demande, veuillez choisir votre moyen de paiement, et accepter les conditions d'utilisation.",
-  "domain_webhosting_order_payment_get_checkout_error": "Une erreur est survenue lors de la récupération des informations de votre panier.",
+  "domain_webhosting_order_payment_get_checkout_error": "Une erreur est survenue lors de la récupération des informations de votre panier : {{ message }}",
+  "domain_webhosting_order_payment_get_checkout_error_unknown": "Une erreur est survenue lors de la récupération des informations de votre panier : {{ message }}",
   "domain_webhosting_order_payment_get_payment_method_error": "Une erreur est survenue lors de la récupération des moyens de paiement.",
-  "domain_webhosting_order_payment_checkout_error": "Une erreur est survenue lors du paiement de votre panier.",
+  "domain_webhosting_order_payment_checkout_error": "Une erreur est survenue lors du paiement de votre panier: {{ message }}",
```
Comment (FredericEspiau) [inline packages/manager/apps/web/client/app/domain/webhosting/order/steps/translations/Messages_fr_FR.json]:
```suggestion
  "domain_webhosting_order_payment_checkout_error": "Une erreur est survenue lors du paiement de votre panier : {{ message }}",
```

File: packages/manager/apps/web/client/app/domain/webhosting/order/steps/domain-webhosting-order-steps.controller.js [hunk 4]
```diff
@@ -1,4 +1,10 @@
 export default class {
+  /* @ngInject */
+
```
Comment (FredericEspiau) [inline packages/manager/apps/web/client/app/domain/webhosting/order/steps/domain-webhosting-order-steps.controller.js]: WS

File: packages/manager/apps/web/client/app/domain/webhosting/order/steps/domain-webhosting-order-steps.controller.js [hunk 5]
```diff
@@ -8,8 +14,12 @@ export default class {
     this.validatingCheckout = true;
     return this.validateCheckout(this.cartId, {
       autoPayWithPreferredPaymentMethod: this.autoPayWithPreferredPaymentMethod,
-    }).catch(() =>
-      this.alertCheckoutError('domain_webhosting_order_payment_checkout_error'),
-    );
+    }).catch(() => {
+      return this.displayErrorMessage(
```
Comment (FredericEspiau) [inline packages/manager/apps/web/client/app/domain/webhosting/order/steps/domain-webhosting-order-steps.controller.js]: This is weird, looks like ESLint or Prettier wasn't triggered on this file

Comment (marie-j) [reply packages/manager/apps/web/client/app/domain/webhosting/order/steps/domain-webhosting-order-steps.controller.js]: Rerun `yarn run lint:js` and no diff was created

**Comment by antleblanc** [comment]
@marie-j Internal ticket DTRSD-6830 is canceled.

Could you please update the Pull Request status (if needed)
and also address the conflict.

Thanks!

**Comment by marie-j** [comment]
> @marie-j Internal ticket DTRSD-6830 is canceled.
> 
> Could you please update the Pull Request status (if needed)
> and also address the conflict.
> 
> Thanks!

@antleblanc Yes this issue was noticed with the ticket but is not solving it.

File: packages/manager/apps/web/client/app/domain/webhosting/order/steps/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -105,9 +105,11 @@
   "domain_webhosting_order_payment_duration": "Durée :",
   "domain_webhosting_order_duration_P1Y": "12 mois",
   "domain_webhosting_order_payment_choice": "Afin de confirmer votre demande, veuillez choisir votre moyen de paiement, et accepter les conditions d'utilisation.",
-  "domain_webhosting_order_payment_get_checkout_error": "Une erreur est survenue lors de la récupération des informations de votre panier.",
+  "domain_webhosting_order_payment_get_checkout_error": "Une erreur est survenue lors de la récupération des informations de votre panier : {{ message }}",
+  "domain_webhosting_order_payment_get_checkout_error_unknown": "Une erreur est survenue lors de la récupération des informations de votre panier : {{ message }}",
```
Comment (JeremyDec) [inline packages/manager/apps/web/client/app/domain/webhosting/order/steps/translations/Messages_fr_FR.json]: I guess you want 
```suggestion
  "domain_webhosting_order_payment_get_checkout_error_unknown": "Une erreur est survenue lors de la récupération des informations de votre panier.",
```

**Pull request merged**